### ISSUE TITLE
Missing semicolon not detected when followed by a nested block

### ISSUE THREAD

**Issue by kevinkorngut** [stylelint/stylelint#1346]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I am not receiving an error when I omit a semicolon.

> Which rule, if any, is this issue related to?

This might relate to what was done to fix issue #1153 

> What CSS is needed to reproduce this issue?

```css
.hello-world {
  max-width: 95vw

  & div {
    display: block;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

```
{
  extends: [
  ],
  rules: {
    "declaration-block-trailing-semicolon": "always"
  }
}
```

> Which version of stylelint are you using?

`6.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI `stylelint '*.css' --verbose`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

It may have to do with the line being followed by a nested block; but I'm not sure.

> What did you expect to happen?

To see an error

> What actually happened (e.g. what warnings or errors you are getting)?

No problems were found.

**Comment by jeddy3** [comment]
As discovered in https://github.com/stylelint/stylelint/issues/1347, PostCSS parses the declaration (with the missing semicolon) as part of the succeeding rule selector.

Anyone got any ideas how we should go about addressing this?

**Comment by davidtheclark** [comment]
Tough, one, because `95vw\n\n  & div` *could* be a selector, and we wouldn't want to penalize somebody if it *were* in their case. 

@kevinkorngut How does that code compile? Into what you want, or what you don't want? (If it compiles as desired, maybe postcss-nested has a solution to this.)

**Comment by kevinkorngut** [comment]
@davidtheclark ok, so using the aforementioned css, missing the semicolon after 95vw, and the following versions:

  "devDependencies": {
    "postcss": "5.0.21",
    "postcss-cli": "2.5.2",
    "postcss-cssnext": "2.5.2",
    "postcss-nested": "1.0.0",
    "stylelint": "6.5.1"
  }

with `postcss-nested` it does not return what I would expect:

```shell
postcss --use postcss-nested test.css
```
yields:

```css
max-width: 95vw

  .hello-world div {
    display: block;
}
```

which although it grabs the property/value for the max-width it loses the selector; and I'd try `postcss-cssnext` but it is using `postcss-selector-parser` and it can't handle the missing semicolon which gives the `Error: Unexpected "space" found` which was goes back to#1153.

**Comment by jeddy3** [comment]
@davidtheclark I'm beginning to think resolving this isn't feasible. Shall we close this issue by adding a note to the rule's README?

**Comment by davidtheclark** [comment]
@jeddy3 Yes.

If any Future Person runs across this issue, we're open to a PR fixing it, if you can figure out a good way.
